runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/AccumulatorSingleton.java

```diff
@@ -69,10 +71,14 @@
   private static void recoverValueFromCheckpoint(JavaSparkContext jsc, String checkpointDir) {
     FSDataInputStream is = null;
     try {
-      checkpointPath = new Path(checkpointDir, ACCUMULATOR_CHECKPOINT_FILE);
+      Path beamCheckpointPath = new Path(checkpointDir, BEAM_CHECKPOINT_DIR);
+      checkpointPath = new Path(beamCheckpointPath, ACCUMULATOR_CHECKPOINT_FILENAME);
       tempCheckpointPath = checkpointPath.suffix(".tmp");
       backupCheckpointPath = checkpointPath.suffix(".bak");
       fileSystem = checkpointPath.getFileSystem(jsc.hadoopConfiguration());
+      if (!fileSystem.exists(beamCheckpointPath)) {
```

Comment:
Beam (maybe more accurate to call it `Runner`) checkpoint path should be created on startup, if it doesn't exist something is broken... maybe throw ?